src/Textfield.js

```diff
@@ -60,15 +52,19 @@ class Textfield extends React.Component {
         var inputId = 'textfield-' + label.replace(/[^a-z0-9]/gi, '');
         var inputTag = hasRows || maxRows > 1 ? 'textarea' : 'input';
 
-        var input = React.createElement(inputTag, {
+        var inputProps = {
             className: classNames('mdl-textfield__input', inputClassName),
             id: inputId,
             key: inputId,
             value: value,
-            onChange: this._handleChange,
             rows: rows,
             ...otherProps
-        });
+        };
+
+        if (onChange) inputProps.onChange =
+            event => this.props.onChange(event.target.value);
```

Comment:
Sorry, I just got used to use `react-hot-loader` and now `react-transform-hmr`. So I avoid assigning methods on each instance.